src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+ * @file linear_svm_main.cpp
+ * @author Yashwant Singh Parihar
+ *
+ * Main executable for linear svm.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+
+#include "linear_svm.hpp"
+
+#include <ensmallen.hpp>
+
+using namespace std;
+using namespace mlpack;
+using namespace mlpack::svm;
+using namespace mlpack::util;
+
+PROGRAM_INFO("LinearSVM is an L2-regularized support vector machine model",
+    // Short description.
+    "An implementation of LinearSVM for multiclass classification. "
+    "Given labeled data, a model can be trained and saved for "
+    "future use; or, a pre-trained model can be used to classify new points.",
+    // Long description.
+    "An implementation of LinearSVM using either the "
+    "L-BFGS optimizer or ParallelSGD (stochastic gradient descent)."
+    "  This solves the classification problem."
+    "\n\n"
+    "This program allows loading a LinearSVM model (via the " +
+    PRINT_PARAM_STRING("input_model") + " parameter) "
+    "or training a LinearSVM model given training data (specified "
+    "with the " + PRINT_PARAM_STRING("training") + " parameter), or both "
+    "those things at once.  In addition, this program allows classification on "
+    "a test dataset (specified with the " + PRINT_PARAM_STRING("test") + " "
+    "parameter) and the classification results may be saved with the " +
+    PRINT_PARAM_STRING("predictions") + " output parameter."
+    " The trained LinearSVM model may be saved using the " +
+    PRINT_PARAM_STRING("output_model") + " output parameter."
+    "\n\n"
+    "The training data, if specified, may have class labels as its last "
+    "dimension.  Alternately, the " + PRINT_PARAM_STRING("labels") + " "
+    "parameter may be used to specify a separate matrix of labels."
+    "\n\n"
+    "When a model is being trained, there are many options.  L2 regularization "
+    "(to prevent overfitting) can be specified with the " +
+    PRINT_PARAM_STRING("lambda") + " option, and The number of classes can be "
+    "manually specified with the " + PRINT_PARAM_STRING("number_of_classes") +
+    "and if an intercept term is not desired in the model, the " +
+    PRINT_PARAM_STRING("no_intercept") + " parameter can be specified."
+    "Margin of difference between correct class and other classes can "
+    "be specified with the" + PRINT_PARAM_STRING("delta") + "option."
+    "The optimizer used to train the model can be specified with the " +
+    PRINT_PARAM_STRING("optimizer") + " parameter.  Available options are "
+    "'psgd' (stochastic gradient descent) and 'lbfgs' (the L-BFGS optimizer).  "
```

Comment:
It might be worth writing `parallel stochastic gradient descent` not just `stochastic gradient descent` here since there are lots of variations of SGD.